metron-platform/metron-parsers/src/main/java/org/apache/metron/parsers/interfaces/MessageParser.java

```diff
@@ -34,15 +34,15 @@
    * @param rawMessage
    * @return If null is returned, this is treated as an empty list.
    */
-  List<T> parse(byte[] rawMessage);
+  List<T> parse(byte[] rawMessage, SensorParserConfig sensorParserConfig);
```

Comment:
Now that I think about it, you could actually do the following:
* Provide a `configurationChanged` method in `MessageParser` with a default noop implementation
* Change the argument to `parseOptionally` to take a Config and in the default implementation of `parseOptionally` call `configurationChanged` upon configuration change.  

This would leave `parse` alone and provide the capability to reconfigure on configuration changes seamlessly to parsers going forward.